README.md

```diff
@@ -143,69 +264,114 @@ Implementing push notifications requires a few additional snippets of code to en
-	if !handled {
-	   // not a klaviyo notification should be handled by other app code
-	}
-    }
-}
+<details>
+<summary> Cocoapods </summary>
 
-```
+- In your `Podfile` add in `KlaviyoSwiftExtension` as a dependency to the newly added notification service extension target.
 
-Once your first push notifications are sent and opened, you should start to see _Opened Push_ metrics within your Klaviyo dashboard.
+  Example:
 
-### Foreground push handling
+  ```
+  target 'NotificationServiceExtension' do
+      pod 'KlaviyoSwiftExtension', '2.1.0-beta1'
+  end
+  ```
 
-The following code example allows push notifications to be displayed when your app is running:
+  Be sure to replace the name of your notification service extension target above.
 
-```swift
+  - Once you've added in the dependency make sure to `pod install`.
+  - Then in the `NotificationService.swift` file add the code for the two required delegates from [this](Examples/KlaviyoSwiftExamples/CocoapodsExample/NotificationServiceExtension/NotificationService.swift) file. This sample covers calling into Klaviyo so that we can download and attach the media to the push notification.
+</details>
 
-    func userNotificationCenter(_ center: UNUserNotificationCenter,
-                                  willPresent notification: UNNotification,
-                                  withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
-        var options: UNNotificationPresentationOptions =  [.alert]
-        if #available(iOS 14.0, *) {
-          options = [.list, .banner]
-        }
-        completionHandler(options)
-    }
-```
+* Test your rich push notifications
+  * Any push notifications tester like apple official [push notification console](https://developer.apple.com/notifications/push-notifications-console/) or a third party software such as [this](https://github.com/onmyway133/PushNotifications).
+  * A push notification payload that resembles what Klaviyo would send to you. The below payload should work as long as the image is valid:
 
-If a user taps on the notification with the application open, this event is tracked as an _Opened Push_ event.
+    ```json
+    {
+    "aps": {
+    "alert": {
+      "title": "Free apple vision pro",
+      "body": "Free Apple vision pro when you buy a Klaviyo subscription."
+    },
+    "mutable-content": 1
+    },
+    "rich-media": "https://www.apple.com/v/apple-vision-pro/a/images/overview/hero/portrait_base__bwsgtdddcl7m_large.jpg",
+    "rich-media-type": "jpg"
+    }
+    ```
+  * A real device's push notification token. This can be printed out to the console from the `didRegisterForRemoteNotificationsWithDeviceToken` method in `AppDelegate`.
+  * Once we have these three things we can then use the push notifications tester and send a local push notification to make sure that everything was set up correctly.
```

Comment:
It feels like we should change "we" to "you." "Once you have these three things, you can then use the push notifications tester and send a local push notification to make sure that everything was set up correctly."

Reply:
Done.